coalib/bears/Bear.py

```diff
@@ -274,6 +281,284 @@ def log_message(self, log_message, timestamp=None, **kwargs):
+                for line in table_header:
+                    print(line)
+                cprint(table.table)
+        finally:
+            remove(path)
+
+    def setup_profiler(self, *args, **kwargs):
+        profile_output =kwargs.get('profile_output', False)
+        profile_dump = kwargs.get('profile_dump', False)
+        profile_commands = kwargs.get('profile_commands', False)
+        if str(profile_output) is not u'False':
+            profile_bears = self.parenthesis_split(str(profile_output))
+        if str(profile_dump) is not u'False':
+            profile_bears = self.parenthesis_split(str(profile_dump))
+        if str(profile_commands) is not u'False':
+            profile_bears = self.parenthesis_split(str(profile_commands))
+        if (str(profile_output) is u'False') and (str(profile_dump) is u'False') and (str(profile_commands) is u'False'):
+            kwargs.pop('section_name')
+            kwargs.pop('bear_name')
+            kwargs.pop('profile_output')
+            kwargs.pop('profile_dump')
+            kwargs.pop('profile_commands')
+            return False, args, kwargs
+        if str(profile_dump) is not u'False' :
+            kwargs.pop('profile_dump')
+            kwargs.pop('profile_output')
+            kwargs.pop('profile_commands')
+            path =  os.getcwd()
+            if str(profile_dump)==u'True':
+                filename = '{}_{}.prof'.format(
+                    kwargs['section_name'], kwargs['bear_name'])
+                open(filename, 'w+')
+                kwargs.pop('section_name')
+                kwargs.pop('bear_name')
+                cProfile.runctx('self.run(*args, **kwargs)',
+                                globals(), locals(), filename)
+            else:
+                subdir = str(profile_dump)
+                filepath = os.path.join(path, subdir)
+                if os.path.isdir(filepath):
+                    os.chdir(subdir)
+                else:
+                    os.mkdir(os.path.join(path, subdir))
+                    os.chdir(subdir)
+                filename = '{}_{}.prof'.format(
+                    kwargs['section_name'], kwargs['bear_name'])
+                open(filename, 'w+')
+                kwargs.pop('section_name')
+                kwargs.pop('bear_name')
+                cProfile.runctx('self.run(*args, **kwargs)',
+                                globals(), locals(), filename)
+                os.chdir("..")
+            return False, args, kwargs
+        if str(profile_output) is not u'False':
+            str_profile_bears = str(profile_bears[0]).lower().strip()
+            prof = cProfile.Profile()
+            kwargs.pop('section_name')
+            kwargs.pop('bear_name')
+            kwargs.pop('profile_dump')
+            kwargs.pop('profile_commands')
+            if not str_profile_bears == u'false':
+                if (u'wrapping_function of' in str(self.run)) or (u'_new_func of' in str(self.run)):
```

Comment:
Line is longer than allowed. (100 > 80)

*Origin: LineLengthBear, Section: `linelength`.*